homeassistant/config.py

```diff
@@ -1023,12 +1035,36 @@ async def merge_packages_config(
     config: dict,
     packages: dict[str, Any],
     _log_pkg_error: Callable[
-        [HomeAssistant, str, str, dict, str], None
+        [HomeAssistant, str, str | None, dict, str], None
     ] = _log_pkg_error,
 ) -> dict:
-    """Merge packages into the top-level configuration. Mutate config."""
+    """Merge packages into the top-level configuration. Ignores packages that cannot be setup. Mutates config.
+
+    Raises vol.Invalid if whole package config is invalid.
+    """
+
+    def _validate_package_definition(name: str, conf: Any) -> None:
```

Comment:
This function can be moved out to the module level since it doesn't depend on any local context.

Reply:
Yes, but from a code readability perspective I think it's beneficial to keep it as an inner function / closure.

Reply:
See, I don't really care about one or the other; it's just unclear what to do now when reviewers don't agree on a topic 🤷 

I moved the function def up a couple of lines now; there _is_ precedence for this in the same file (several functions could have been defined as inner functions but weren't)